pkg/rules/rules.go

```diff
@@ -58,6 +61,16 @@ func (rr *GRPCClient) Rules(ctx context.Context, req *rulespb.RulesRequest) (*ru
 		return nil, nil, errors.Wrap(err, "proxy Rules")
 	}
 
+	var err error
+	matcherSets := make([][]*labels.Matcher, len(req.MatcherString))
```

Comment:
should we check if `len(req.MatcherString) > 0` or this is already covered from [here](https://github.com/thanos-io/thanos/pull/5111/files#diff-f14ed7ae9f9be7d76dea77aa31304b22de00af7e5f08720a7b6e75f721caf24aR790)?

Reply:
If the length is zero, `matcherSets` would be an empty array of type `[][]*labels.Matcher` and the loop below this won't run. `filterRules` gets called right after and it checks the length of `matcherSets` [here](https://github.com/thanos-io/thanos/pull/5111/files#diff-a46b946d0da6144f5d8f5515bf381041db57211dff8fec08daf39fc46ab2b0a1R85) and if it is 0, returns all the rules. So we kind of do check for the length of `MatcherString`, just not directly. 🙂

Reply:
sounds good!